File: src/main/java/com/sleepkqq/sololeveling/localization/LocaleConverter.java
package com.sleepkqq.sololeveling.localization;

import jakarta.persistence.AttributeConverter;
import jakarta.persistence.Converter;

import java.util.Locale;

@Converter(autoApply = true)
public class LocaleConverter implements AttributeConverter<Locale, String> {

  @Override
  public String convertToDatabaseColumn(Locale locale) {
    return locale == null ? null : locale.toLanguageTag();
  }

  @Override
  public Locale convertToEntityAttribute(String dbData) {
    return dbData == null ? null : Locale.forLanguageTag(dbData);
  }
}

File: solo-leveling-gateway-service/src/main/kotlin/com.sleepkqq.sololeveling/gateway/localization/LocaleConverter.kt
package com.sleepkqq.sololeveling.gateway.localization

import jakarta.persistence.AttributeConverter
import jakarta.persistence.Converter
import java.util.Locale

@Converter(autoApply = true)
class LocaleConverter : AttributeConverter<Locale, String> {

	override fun convertToDatabaseColumn(locale: Locale): String = locale.toLanguageTag()

	override fun convertToEntityAttribute(dbData: String): Locale =
		Locale.forLanguageTag(dbData)
}
